### PR TITLE
fixes in create function apps

### PR DESCRIPTION
**Pull request by aspaan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -171,6 +171,6 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
 register_cli_argument('functionapp create', 'plan', options_list=('--plan', '-p'), completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
                       help="name or resource id of the function app service plan. Use 'appservice plan create' to get one")
 register_cli_argument('functionapp create', 'consumption_plan_location', options_list=('--consumption-plan-location', '-c'),
-                      help="Geo location for functionsapps with consumption plans. Use 'functionapp list-consumption-locations' to see possible location plans")
+                      help="Geo location for functionsapps with consumption plans. Use 'functionapp list-consumption-locations' to see possible locations")
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: Does it make sense to say `Geo locations with consumption plans available you can create functionapps from...`. I am flexible on wording as long as the help communicates this `location` is meant to use like a `plan`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -964,43 +964,44 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
 def create_function(resource_group_name, name, storage_account, plan=None,
                     consumption_plan_location=None):
 
-    if plan is not None and consumption_plan_location is not None:
-        raise CLIError("Use Plan or Consumption Plan Location")
+    if ((plan and consumption_plan_location) or
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: you can simplify as `bool(plan) == bool(consumption_plan_location )`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -964,43 +964,44 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
...
+            (plan is None and consumption_plan_location is None)):
+        raise CLIError("Use either Plan or Consumption Plan Location")
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: please use the error Travis suggested in the previous PR `usage error: --plan NAME_OR_ID | --consumption-plan-location LOCATION`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 4]
```diff
@@ -1029,8 +1030,8 @@ def _validate_and_get_connection_string(resource_group_name, storage_account):
         raise CLIError(error_message)
 
     keys = storage_client.storage_accounts.list_keys(resource_group_name, storage_account).keys
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Would it make sense consolidate [error reporting](https://github.com/Azure/azure-cli/blob/functionapp/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py#L1019-L1026)  to code like:
```python
    for e in ['blob', 'queue', 'table']:
        if not getattr(endpoints, e, None):
              error_message = "Storage account has no '{}' endpoint. It must have table, queue, and blob endpoints all enabled".format(e)
```

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py (hunk 1 above)
Comment (lindydonna) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: I would say 

> Geographic location where Function App will be hosted. Use 'functionapp list-consumption-locations' to view available locations.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 4 above)
Comment (lindydonna) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Slight change to wording:
> Storage account has no '{}' endpoint. A storage account that supports blob, queue, and table storage is required.

Even better:
Storage account 'name' has no '{}' endpoint...

**Comment by SyntaxC4** [comment]
@yugangw-msft Feedback addressed and CI building, can you please merge at earliest convenience. I have the Bug Bash email ready to go.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 4 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: @aspaan, in the next PR, please update the message like Donna suggested, or you can choose to do it when she goes through all of the text

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 5]
```diff
@@ -171,6 +171,10 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
 register_cli_argument('functionapp create', 'plan', options_list=('--plan', '-p'), completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
                       help="name or resource id of the function app service plan. Use 'appservice plan create' to get one")
 register_cli_argument('functionapp create', 'consumption_plan_location', options_list=('--consumption-plan-location', '-c'),
-                      help="Geo location for functionsapps with consumption plans. Use 'functionapp list-consumption-locations' to see possible location plans")
+                      help="Geographic location where Function App will be hosted. Use 'functionapp list-consumption-locations' to view available locations.")
 register_cli_argument('functionapp create', 'storage_account', options_list=('--storage-account', '-s'),
                       help='Provide a string value of a Storage Account in the provided Resource Group. Or Resource ID of a Storage Account in a different Resource Group')
+
+register_cli_argument('functionapp config appsettings', 'settings', nargs='+', help="space separated app settings in a format of <name>=<value>")
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: @aspaan, in the next PR, please do Travis [suggested in the previous PR ](https://github.com/Azure/azure-cli/pull/2829/files#r110995725)

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py (hunk 1 above)
Comment (johnpapa) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: I recommend being explicit in help commands and including the `az` command. FOr example, change this to ... 

```
help="Geo location for functionsapps with consumption plans. Use 'az functionapp list-consumption-locations' to see possible locations")
```

This allows people to copy and paste easily, and not guess.

Same would go for all places in help.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 6]
```diff
@@ -171,6 +171,10 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
 register_cli_argument('functionapp create', 'plan', options_list=('--plan', '-p'), completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
                       help="name or resource id of the function app service plan. Use 'appservice plan create' to get one")
 register_cli_argument('functionapp create', 'consumption_plan_location', options_list=('--consumption-plan-location', '-c'),
-                      help="Geo location for functionsapps with consumption plans. Use 'functionapp list-consumption-locations' to see possible location plans")
```
Comment (johnpapa) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: I recommend being explicit in help commands and including the az command. For example, change this to ...

`help="Geo location for functionsapps with consumption plans. Use 'az functionapp list-consumption-locations' to see possible location plans"`

**Pull request merged**